types/stylelint/index.d.ts

```diff
@@ -312,10 +326,14 @@ declare namespace stylelint {
 			{ ignore: OneOrMany<'with-var-inside'> }
 		>;
 		'color-hex-alpha': CoreRule<'always' | 'never'>;
-		'color-hex-length': CoreRule<'short' | 'long'>;
+		'color-hex-length': CoreRule<'short' | 'long', {}, AutofixMessage>;
 		'color-named': CoreRule<
 			'never' | 'always-where-possible',
-			{ ignoreProperties: OneOrMany<StringOrRegex>; ignore: OneOrMany<'inside-function'> }
+			{ ignoreProperties: OneOrMany<StringOrRegex>; ignore: OneOrMany<'inside-function'> },
+			{
+				expected: (expected: string, actual: string) => string;
+				rejected: (keyword: string) => string;
```

Comment:
[suggestion] How about changing the argument name in the implementation to align this type addition?

https://github.com/stylelint/stylelint/blob/3a2f6585340c5a927248ee1c12b77dff2404b63c/lib/rules/color-named/index.mjs#L22

```diff
-	rejected: (named) => `Unexpected named color "${named}"`,
+	rejected: (keyword) => `Unexpected named color "${keyword}"`,
```

Reply:
The goal is to have hints that are standardised so that we may reuse those terms later on.
i.e. `named` won't be reused by other rules
e.g. `declaration-property-value-keyword-no-deprecated` will also use the term keyword

It has to specific enough as a hint for the developer but not too specific so that it can part of a shortlist of terms used for the names of the arguments.
It will become clear once we add all third argument to the rules that have a `messageArgs` returned.
i.e. I didn't want this PR to be too long to review